install/migrations/update_10.0.6_to_10.0.7/configs.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var Migration $migration
+ */
+
+$migration->displayMessage('Add new configurations / user preferences');
+$migration->addConfig([
+    'use_flat_dropdowntree_on_search_result'   => 0,
+]);
+$migration->addField('glpi_users', 'use_flat_dropdowntree_on_search_result', 'tinyint DEFAULT 0');
```

Comment:
User preferences should have the NULL value by default. This value indicates that user has not set its preference and will use the value defined in global config.

```suggestion
$migration->addField('glpi_users', 'use_flat_dropdowntree_on_search_result', 'tinyint DEFAULT NULL');
```